File: src/Week1/pset1/cash.c
#include <cs50.h>
#include <math.h>
#include <stdio.h>

const int QUARTERS = 25;
const int DIMES = 10;
const int NICKELS = 5;
const int PENNIES = 1;

int number_of_coins(int cents);

int main(void)
{
    float n = 0;
    do
    {
        n = get_float("Change owed: ");
    }
    while (n < 0);
    int cents = round(n * 100);
    printf("%i\n", number_of_coins(cents));
}

//Returns the number of coins needed to give the change
int number_of_coins(int cents)
{
    int coins = 0;
    while (cents != 0)
    {
        if (cents - QUARTERS >= 0)
        {
            cents -= QUARTERS;
            coins++;
        }
        else if (cents - DIMES >= 0)
        {
            cents -= DIMES;
            coins++;
        }
        else if (cents - NICKELS >= 0)
        {
            cents -= NICKELS;
            coins++;
        }
        else
        {
            cents -= PENNIES;
            coins++;
        }
    }
    return coins;
}

File: src/Week6/pset6/cash/cash.py
from cs50 import get_float

while True:
    change = get_float("Change owed: ")
    if change >= 0:
        break
cents = change * 100
coins = 0

while cents > 0:
    if (cents - 25) >= 0:
        cents -= 25
    elif (cents - 10) >= 0:
        cents -= 10
    elif (cents - 5) >= 0:
        cents -= 5
    else:
        cents -= 1
    coins += 1

print(coins)
